### ISSUE TITLE
charmcraft 2.5.2 on PyPi doesn't install on jammy/py310 (at least) - error "ModuleNotFoundError: No module named 'tools'"

### ISSUE THREAD

**Issue by ajkavanagh** [canonical/charmcraft#1472]
### Bug Description

A tox env with charmcraft >= 2.5.0 fails to install with charmcraft:

```
Collecting charmcraft<3.0 (from -r /home/alex/Projects/Canonical/git/github.com/openstack-charmers/charmhub-lp-tools/requirements.txt (line 16))
  Using cached charmcraft-2.5.2.tar.gz (185 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "/home/alex/Projects/Canonical/git/github.com/openstack-charmers/charmhub-lp-tools/.tox/py3/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/alex/Projects/Canonical/git/github.com/openstack-charmers/charmhub-lp-tools/.tox/py3/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/alex/Projects/Canonical/git/github.com/openstack-charmers/charmhub-lp-tools/.tox/py3/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-8bdjwu91/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-8bdjwu91/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-8bdjwu91/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 480, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-8bdjwu91/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 23, in <module>
      ModuleNotFoundError: No module named 'tools'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
```

### To Reproduce

Create a virtualenv.

```bash
virtualenv -p `which python3` venv
. venv/bin/activate
pip install charmcraft
```

### Environment

Ubuntu Jammy (22.04)

```
$ python --version
Python 3.10.12
```

### charmcraft.yaml

```shell
N/A
```

### Relevant log output

```shell
alex@alex-x570: ~/tmp/tempvenv ❯ virtualenv -p `which python3` venv
created virtual environment CPython3.10.12.final.0-64 in 126ms
  creator CPython3Posix(dest=/home/alex/tmp/tempvenv/venv, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/alex/.local/share/virtualenv)
    added seed packages: pip==23.2, setuptools==68.0.0, wheel==0.40.0
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator
alex@alex-x570: ~/tmp/tempvenv ❯ . venv/bin/activate
alex@alex-x570: ~/tmp/tempvenv 🐍 (venv)❯ pip install charmcraft
Collecting charmcraft
  Using cached charmcraft-2.5.2.tar.gz (185 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "/home/alex/tmp/tempvenv/venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/alex/tmp/tempvenv/venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/alex/tmp/tempvenv/venv/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-75mqadoz/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-75mqadoz/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-75mqadoz/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 480, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-75mqadoz/overlay/lib/python3.10/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 23, in <module>
      ModuleNotFoundError: No module named 'tools'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```
```

**Comment by bittner** [comment]
I can confirm this problem. It looks like a packaging error. The [`tools` module](https://github.com/canonical/charmcraft/tree/main/tools) is not included [in the `find` directive](https://github.com/canonical/charmcraft/blob/main/pyproject.toml#L105).

**Comment by lengau** [comment]
This was fixed in Charmcraft 3.0. Does it need backporting to 2.x?
